src/3d/qgsrulebasedchunkloader_p.cpp

```diff
@@ -166,7 +166,7 @@ QgsRuleBasedChunkLoaderFactory::QgsRuleBasedChunkLoaderFactory( const Qgs3DMapSe
   , mRootRule( rootRule->clone() )
   , mLeafLevel( leafLevel )
 {
-  const QgsAABB rootBbox = Qgs3DUtils::layerToWorldExtent( vl->extent(), zMin, zMax, vl->crs(), map.origin(), map.crs(), map.transformContext() );
+  const QgsAABB rootBbox = Qgs3DUtils::mapToWorldExtent( map.extent(), zMin, zMax, map.origin() );
```

Comment:
It seems to me that `map.extent()` can be an empty rectangle - would this work in such case?

Reply:
`map.extent()` should not be empty, it should always be set. Even when the whole project is used for a 3d scene, it should have that set as the project's extent.
This is relevant to the other map.extent() comments and might need some more though regarding _how_ this extent will be set. Right now it is only used by `Qgs3DMapSettings::configureTerrainFromProject()` itself, hence no need to emit any signal.